test/run

```diff
@@ -2,6 +2,14 @@
 
 set -eu
 
+test_ruff() {
+    command -v ruff >/dev/null || echo 'no ruff installed'
+    test -n "$(ruff check --show-files . 2>/dev/null)" || return 0  # no python changes
```

Comment:
I don't like suppressing stderr here. Is that just for the case that ruff isn't installed? (see above)